### DISCUSSION TITLE
Transforming returned data from BigQuery

### DISCUSSION THREAD

**Discussion by AndyGUK** [Budibase/budibase#5113] [Help] [answered]
I'm hoping someone might be able to help out me transform my data using Budibase. I think I've followed the steps detailed in the [BigQuery documentation](https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/list?apix_params=%7B%22projectId%22%3A%22guk-etl-327714%22%2C%22datasetId%22%3A%22marketing_hub%22%2C%22tableId%22%3A%22channels%22%7D) correctly, now I just need to transform the data into a usable format.

This is my result at the moment.
![image](https://user-images.githubusercontent.com/73335367/160242283-8974f490-c913-442c-b9fc-ff1463f562d9.png)

I've tried using something like `return data.rows` but that creates the following output:
![image](https://user-images.githubusercontent.com/73335367/160242338-0defda18-e7fc-4c55-8815-23dab78d7f3e.png)

which will look something like:
```
{
  "f": [
    {
      "v": "23850158465680196.0"
    },
    {
      "v": "139"
    },
    {
      "v": "facebook"
    }
  ]
}
```

I think I potentially need to use a for loop in order to separate the values out of the fields to create a record with three columns:
channel_id | campaign_id | franchise_id

Is there an easy way to do this? My JavaScript skills are lacking but since there's no BigQuery data connector this is the only way.

**Comment by shogunpurple** [comment, marked as answer]
Hi @AndyGUK - if you can always rely on the order of the array values to be `channel_id | campaign_id | franchise_id`, and you only have one flat `f` row in your bigquery response as you say:

To transform this:
```
{
  "rows": [{
    "f": [{
        "v": "23850158465680196.0"
      },
      {
        "v": "139"
      },
      {
        "v": "facebook"
      }
    ]
  }]
}
```

into this:
```
[{
  "channel_id": "23850158465680196.0",
  "campaign_id": "139",
  "franchise_id": "facebook",
}]
```

You can use the following transformer:

```js
const transformed = data.rows.map(row => {
  const [channel_id, campaign_id, franchise_id] = row.f
  return {
    channel_id: channel_id.v,
    campaign_id: campaign_id.v,
    franchise_id: franchise_id.v
  }
})

return transformed
```

Hope this helps. Thanks!

**Reply by AndyGUK** [reply]
Hi @shogunpurple, this works a treat! Thank you so much! This is a real help as I can hopefully soon start working with more of my data.

Appreciate the help!
